tests/unit/SidebarOrderTest.ts

```diff
@@ -63,40 +63,38 @@ describe('Sidebar', () => {
             expect(screen.toJSON()).toBe(null);
         });
 
-        it('is rendered with an empty list when personal details exist', () => {
-            // Given the sidebar is rendered with default props
-            LHNTestUtils.getDefaultRenderedSidebarLinks();
-
-            return (
-                waitForBatchedUpdates()
-                    // When Onyx is updated with some personal details
-                    .then(() =>
-                        Onyx.multiSet({
-                            [ONYXKEYS.PERSONAL_DETAILS_LIST]: LHNTestUtils.fakePersonalDetails,
-                            [ONYXKEYS.IS_LOADING_APP]: false,
-                        }),
-                    )
-
-                    // Then the component should be rendered with an empty list since it will get past the early return
-                    .then(() => {
-                        expect(screen.toJSON()).not.toBe(null);
-                        const navigatesToChatHintText = Localize.translateLocal('accessibilityHints.navigatesToChat');
-                        expect(screen.queryAllByAccessibilityHint(navigatesToChatHintText)).toHaveLength(0);
-                    })
-            );
-        });
+        it('is rendered with an empty list when personal details exist', () =>
+            waitForBatchedUpdates()
+                // Given the sidebar is rendered with default props
+                .then(() => LHNTestUtils.getDefaultRenderedSidebarLinks('0'))
```

Comment:
Did the previous tests causing any errors if we using `LHNTestUtils.getDefaultRenderedSidebarLinks` before `waitForBatchedUpdates` finished?

Reply:
Yes, it caused many errors. I think it happens because `withOnyx` works differently than `useOnyx`.